src/client/api/types.ts

```diff
@@ -0,0 +1,47 @@
+import type { BaseProvider } from "./base";
+
+export type ProviderInfo = {
+  /**
+   * full url used for request
+   */
+  url: string;
+  /**
+   * base url of the request
+   */
+  baseUrl: string;
+  /**
+   * route elements which will follow the baseUrl
+   */
+  route: string[];
+  /**
+   * headers to include in the request
+   */
+  appendHeaders: Record<string, string>;
+  /**
+   * provider owner
+   */
+  owner: Owner;
+};
+
+export type ApiKind = "llm" | "email";
+export type Owner = EmailOwner | LLMOwner;
```

Comment:
I feel like `Owner` is not the correct keyword here, because `EmailOwner` or `LLMOwner` feels like referring to a user, not a company. I was thinking of `Provider` at first, but we already use that for classes. Can we think of sth else?

Reply:
I took the owner from the old api. It's not the best name but I couldn't think of a better alternative. Something like `ProviderOwner` would be too much I think